### PR TITLE
V2.x 4729 merge v1.x

### PR DESCRIPTION
**Pull request by Triplkrypl**
https://redmine.havelholding.cz/issues/4729

Basically same change as: https://github.com/ThePay/api-client/pull/52, with resolved merge conflict and updates logic using PSR HTTP client and new version of phpunit.

### CODE REVIEW

File: src/Service/ApiServiceInterface.php [hunk 1]
```diff
@@ -138,4 +138,15 @@ public function createPaymentRefund(Identifier $uid, Amount $amount, string $rea
...
+    /**
+     * Method will generate PDF file as confirmation for paid payment
+     *
+     * @see https://dataapi21.docs.apiary.io/#reference/data-retrieval/payments/get-payment-confirmation
+     *
+     * @return string with binary content of PDF file
+     *
+     * @throws ApiException if payment is not paid yet
+     */
+    public function generatePaymentConfirmationPdf(Identifier $uid, LanguageCode $languageCode = null);
```
Comment (joseftraxler) [inline src/Service/ApiServiceInterface.php]: I recommend using of return types in 2.x vesion as much as you can.

Comment (joseftraxler) [reply src/Service/ApiServiceInterface.php]: As well as parameters.

**Review by joseftraxler — APPROVED** [review]
Nice 👍

**Comment by Triplkrypl** [comment]
@AlexKratky promin za spam notifikace, myslel jsem ve stylu automatu a zapomel ze mas dovcu.

**Review by joseftraxler — APPROVED** [review]
👏

**Pull request merged**